ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,199 @@
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Execute an approved Paystack transaction. This method will record payment processor
+        response for future usage.
+
+        Arguments:
+            response: Transaction response received from Paystack after successfull payment
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Returns:
+            HandledProcessorResponse
+        """
+        authorization = response.get('authorization')
+        transaction_id = response.get('id')
+
+        self.record_processor_response(authorization, transaction_id=transaction_id, basket=basket)
+        logger.info("Successfully executed Paystack payment [%s] for basket [%d].", transaction_id, basket.id)
+
+        currency = response.get('currency')
+        total = float(response.get('amount'))
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=authorization.get('last4'),
+            card_type=authorization.get('card_type')
+        )
+
+    def verify_refund(self, reference_number):
+        """
+        Returns a boolean value indicating refund object status after verification from Paystack
+        """
+        success, response = self.paystack_client.handler(FETCH_REFUND_CODE, reference_number)
+        if success:
+            data = response.get('data')[0]
+            refund_status = data.get('status')
+            logger.info("Paystack refund has been fetched with status: %s ", refund_status)
+
+            if refund_status == 'processed':
+                return True
+
+        logger.error("Unable to fetch refund object from paystack for refund_id: %s", reference_number)
+        return False
+
+    def issue_credit(self, order_number, basket, reference_number, amount, currency):
+        """
+        Execute Paystack refund flow. First we need to create refund object at paystack then need to verify
+        it's status to complete all process.
+
+        Raises:
+            RefundError: indicating general refund error.
+        """
+        try:
+            success, response = self.paystack_client.handler(CREATE_REFUND_CODE, reference_number)
+
+            if success:
+                data = response.get('data')
+                refund_id = data.get('id')
+                if data:
+                    logger.info(
+                        "Successfully created  Paystack refund request for transaction: %s, and got refund_id: %s",
+                        reference_number,
+                        refund_id
+                    )
+                is_verified_refund = self.verify_refund(reference_number)
+                if is_verified_refund:
+                    self.record_processor_response(response, transaction_id=refund_id, basket=basket)
+                    return refund_id
```

Comment:
for consistency can we return `reference_number` here as well.

Reply:
Ideally, we need to return refund_id from here. But I guess it will not make much difference. so I will change it.